### ISSUE TITLE
Plugin causes fatal error when an update is available

### ISSUE THREAD

**Issue by pacmanito** [skyverge/woocommerce-shipping-estimate#27]
Recently (presumably after version 2.3.5 was released) I started getting a fatal error on visiting updates page (https://***.com/wp-admin/update-core.php). As soon as I deactivate this plugin the problem is gone, so I suppose the problem is caused by exactly this plugin. Here's a screenshot:
![image](https://user-images.githubusercontent.com/56888134/88684870-fbeae500-d0fd-11ea-9e5f-4026d208b4d3.png)

**Comment by wvega** [comment]
@pacmanito I'm sorry that the plugin continues to cause problems. Do you have a `fatal-errors` log on WooCommerce > Status > Logs for one of the days when you saw that error?

<img width="1436" alt="image" src="https://user-images.githubusercontent.com/45068/88828526-fa1e3180-d190-11ea-935e-02ca399968b7.png">

If you have one, would you please share that log file with us so that we can see the complete call stack?

If you prefer not to share those details here, you can send us message through the [Contact Us](https://www.skyverge.com/contact/) page and paste the content of the file there. Please select the **I'm getting in touch for something else** in the first dropdown and include a link to this issue too.

Thanks!

**Comment by pacmanito** [comment]
The log is filled with the record shown on error screen:

```
2020-07-28T12:15:29+00:00 CRITICAL Uncaught Error: Cannot use object of type stdClass as array in ***/public_html/wp-admin/update-core.php:337
Stack trace:
#0 ***/public_html/wp-admin/update-core.php(771): list_plugin_updates()
#1 {main}
  thrown in ***/public_html/wp-admin/update-core.php on line 337
```

this record is repeated around dozen times

**Comment by pacmanito** [comment]
Hi, the bug is still there. Now I have WooCommerce Shipping Estimates ver 2.3.5 installed and as soon as version 2.4 became available I started having the same error upon opening update screen. After deactivating your plugin update screen can be opened  without any errors.

**Comment by wvega** [comment]
Hey @pacmanito, I'm sorry the issue is back.

Thank you for providing additional details about what you are seeing. I was finally able to reproduce the problem locally! I misunderstood the issue at first and confused it with another PHP error we were seeing at the time of this report.

While we work on a fix, would you be able to install v2.4.0 to make the error disappear?

**Comment by pacmanito** [comment]
Thanks for quick reply, hope you will be able tio fix the issue. 
I was able to update the plugin through "Installed Plugins" page.

**Comment by unfulvio** [comment]
Hi,

This should have been addressed in latest version (2.4.0) - please let us know if still experiencing the problem.
